hub/saml-engine/src/main/java/uk/gov/ida/hub/samlengine/services/IdpAuthnResponseTranslatorService.java

```diff
@@ -38,24 +40,34 @@
     private final DecoratedSamlResponseToIdaResponseIssuedByIdpTransformer samlResponseToIdaResponseIssuedByIdpTransformer;
     private InboundResponseFromIdpDataGenerator inboundResponseFromIdpDataGenerator;
     private final IdpAssertionMetricsCollector idpAssertionMetricsCollector;
+    private final EidasAttributesLogger eidasAttributesLogger;
+    private final TransactionsConfigProxy transactionsConfigProxy;
 
     @Inject
     public IdpAuthnResponseTranslatorService(StringToOpenSamlObjectTransformer<Response> stringToOpenSamlResponseTransformer,
                                              StringToOpenSamlObjectTransformer<Assertion> stringToAssertionTransformer,
                                                  @Named("IdpSamlResponseTransformer") DecoratedSamlResponseToIdaResponseIssuedByIdpTransformer samlResponseToIdaResponseIssuedByIdpTransformer,
                                              InboundResponseFromIdpDataGenerator inboundResponseFromIdpDataGenerator,
-                                             IdpAssertionMetricsCollector idpAssertionMetricsCollector) {
+                                             IdpAssertionMetricsCollector idpAssertionMetricsCollector,
+                                             @Named("EidasAttributesLogger") EidasAttributesLogger eidasAttributesLogger,
+                                             TransactionsConfigProxy transactionsConfigProxy) {
         this.stringToOpenSamlResponseTransformer = stringToOpenSamlResponseTransformer;
         this.stringToAssertionTransformer = stringToAssertionTransformer;
         this.samlResponseToIdaResponseIssuedByIdpTransformer = samlResponseToIdaResponseIssuedByIdpTransformer;
         this.inboundResponseFromIdpDataGenerator = inboundResponseFromIdpDataGenerator;
         this.idpAssertionMetricsCollector = idpAssertionMetricsCollector;
+        this.eidasAttributesLogger = eidasAttributesLogger;
+        this.transactionsConfigProxy = transactionsConfigProxy;
     }
 
     public InboundResponseFromIdpDto translate(SamlAuthnResponseTranslatorDto samlResponseDto) {
         Response response = stringToOpenSamlResponseTransformer.apply(samlResponseDto.getSamlResponse());
         MdcHelper.addContextToMdc(response);
         try {
+            String matchingServiceEntityId = samlResponseDto.getMatchingServiceEntityId();
+            if (transactionsConfigProxy.isProxyNodeEntityId(matchingServiceEntityId)) {
```

Comment:
Check with `config` whether this `matchingServiceEntityId` is an eidas proxy node service

Reply:
Assuming here the `matchingServiceEntityId` is the same as the transaction entity ID in a non-matching scenario? I don't really get why that `SamlAuthnResponseTranslatorDto` needs to know about the matching service but oh well.

Reply:
Yes, updated PR message to note:

For an eIDAS journey, the matching service entity id will be the request issuer id, as this will be a non-matching journey, see uk/gov/ida/hub/policy/services/AuthnResponseFromIdpService.java:54.
This check occurs in policy, before the response is sent to saml-engine.